plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -458,6 +458,19 @@ func ClusterRoles() []rbac.ClusterRole {
 				eventsRule(),
 			},
 		},
+		{
+			// a role for csi external components {csi-provisioner,csi-attacher,driver-registrar}
+			ObjectMeta: metav1.ObjectMeta{Name: "system:csi-external-components"},
+			Rules: []rbac.PolicyRule{
+				rbac.NewRule("create", "delete", "get", "list", "watch", "update").Groups(legacyGroup).Resources("persistentvolumes").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "update").Groups(legacyGroup).Resources("persistentvolumeclaims").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "create", "update", "patch").Groups(legacyGroup).Resources("events").RuleOrDie(),
+				rbac.NewRule("get", "list").Groups(legacyGroup).Resources("secrets").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "update").Groups(legacyGroup).Resources("nodes").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "update").Groups(storageGroup).Resources("volumeattachments").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch").Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
+			},
+		},
```

Comment:
Should grant patch in addition to update and write external components to use patch to minimize chances of dropping fields on version skewed clusters

Reply:
Right now none of the components use "Patch". It is an open issue and we are aware of it. Should we give "Patch" API access now, or wait until we update the components to support "Patch"ing

Reply:
go ahead and add patch permission, since we know that is the desired operation long-term